File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/CityChain.java
package br.com.alexandre.friday.recognize.chain;

import br.com.alexandre.friday.recognize.RecognizerChain;
import br.com.alexandre.friday.recognize.RecognizerRequest;

import java.util.Arrays;
import java.util.Collection;

public class CityChain extends RecognizerChain {
    public Collection<String> execute(RecognizerRequest request) {
        if (request.getLocality() == null || request.getLocality().trim().isEmpty()) {
            return Arrays.asList(String.format("No momento não consegui verificar a cidade, %s!", request.getName()));
        } else {
            return Arrays.asList(String.format("Você está na cidade de %s", request.getLocality()));
        }
    }

    public boolean apply(RecognizerRequest request) {
        return  (request.getQ().contains("qual") &&
                request.getQ().contains("é") &&
                request.getQ().contains("essa") &&
                request.getQ().contains("cidade"))
                ||
                (request.getQ().contains("que") &&
                        request.getQ().contains("cidade") &&
                        request.getQ().contains("é") &&
                        request.getQ().contains("essa"));
    }
}

File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/CityChain.kt
package br.com.alexandre.friday.recognize.chain

import br.com.alexandre.friday.recognize.RecognizerChain
import br.com.alexandre.friday.recognize.RecognizerRequest
import java.util.*

class CityChain: RecognizerChain() {
    override fun execute(request: RecognizerRequest?): MutableCollection<String> {
        return Arrays.asList(String.format("Você está na cidade de %s", request?.locality))
    }

    override fun apply(request: RecognizerRequest?): Boolean {
        return request?.q!!.contains("qual") &&
                request.q!!.contains("é") &&
                request.q!!.contains("essa") &&
                request.q!!.contains("cidade")
                ||
                request.q!!.contains("que") &&
                request.q!!.contains("cidade") &&
                request.q!!.contains("é") &&
                request.q!!.contains("essa")
    }
}